core/lib/types/src/storage_writes_deduplicator.rs

```diff
@@ -161,23 +190,49 @@ impl StorageWritesDeduplicator {
 
     fn rollback(&mut self, updates: Vec<UpdateItem>) {
         for item in updates.into_iter().rev() {
-            let field_to_change = if item.is_write_initial {
-                &mut self.metrics.initial_storage_writes
+            let (field_to_change, total_size) = if item.is_write_initial {
+                (
+                    &mut self.metrics.initial_storage_writes,
+                    &mut self.metrics.total_writes_size,
```

Comment:
Same here, no in both clausess `&mut self.metrics.total_writes_size,` is returned